Library/Homebrew/cmd/tests.rb

```diff
@@ -36,18 +24,55 @@ def tests
         args << "TESTOPTS=--name=test_#{test_method}" if test_method
       end
       args += ARGV.named.select { |v| v[/^TEST(OPTS)?=/] }
-      system "bundle", "exec", "rake", "test", *args
 
-      Homebrew.failed = !$?.success?
+      begin
+        sanitize_env!
+
+        # Since we "borrow" HOME for duration of tests bundler
+        # needs fetching/installing every run.
+        Homebrew.install_gem_setup_path! "bundler"
+        unless quiet_system("bundle", "check")
+          system "bundle", "install", "--path", "vendor/bundle"
+        end
```

Comment:
I have to admit that I'm not particularly thrilled with this. Do you have any ideas how to avoid this downside? This is going to be annoying when frequently running tests locally (a fetch of the `bundler` Gem for every run) and even worse when trying to do the same while offline, e.g. on a train or plane.